dhall/src/Dhall/Repl.hs

```diff
@@ -398,26 +399,91 @@ cmdQuit _ = do
   liftIO (putStrLn "Goodbye.")
   liftIO (throwIO Interrupt)
 
+help
+  :: ( Haskeline.MonadException m, MonadFail m, MonadIO m, MonadState Env m )
+  => HelpOptions m -> [String] -> m ()
+help hs _ = do
+  liftIO (putStrLn "Welcome to the Dhall REPL.")
+  liftIO (putStrLn "Type any expression to normalize it or use one of the following commands:")
+  forM_ hs $ \h -> do
+    let name = helpOptionName h
+        syntax = helpOptionSyntax h
+        doc = helpOptionDoc h
+    liftIO (putStrLn (":" <> name <> " " <> syntax))
+    liftIO (putStrLn ("    " <> doc))
 
 optionsPrefix :: Char
 optionsPrefix = ':'
 
+data HelpOption m = HelpOption
+  { helpOptionName :: String
+  , helpOptionSyntax :: String
+  , helpOptionDoc :: String
+  , helpOptionFunction :: Repline.Cmd m
+  }
+
+type HelpOptions m = [HelpOption m]
 
-options
+helpOptions
   :: ( Haskeline.MonadException m, MonadFail m, MonadIO m, MonadState Env m )
-  => Repline.Options m
-options =
-  [ ( "type", dontCrash . typeOf )
-  , ( "hash", dontCrash . hashBinding )
-  , ( "let", dontCrash . addBinding . separateEqual )
-  , ( "clear", dontCrash . clearBindings)
-  , ( "load", dontCrash . loadBinding )
-  , ( "save", dontCrash . saveBinding . separateEqual )
-  , ( "set", dontCrash . setOption)
-  , ( "unset", dontCrash . unsetOption)
-  , ( "quit", cmdQuit )
+  => HelpOptions m
+helpOptions =
+  [ HelpOption
+      "help"
+      ""
+      "Print help text and describe options"
+      (dontCrash . help helpOptions)
+  , HelpOption
+      "type"
+      "EXPRESSION"
+      "Infer the type of an expression"
+      (dontCrash . typeOf)
+  , HelpOption
+      "hash"
+      "EXPRESSION"
+      "Hash the normalized value of an expression"
+      (dontCrash . hashBinding)
+  , HelpOption
+      "let"
+      "IDENTIFIER = EXPRESSION"
+      "Assign an expression to a variable"
+      (dontCrash . addBinding . separateEqual)
+  , HelpOption
+      "clear"
+      ""
+      "Clear all bound variables"
+      (dontCrash . clearBindings)
+  , HelpOption
+      "load"
+      "[FILENAME]"
+      "Load bound variables from a file"
+      (dontCrash . loadBinding)
+  , HelpOption
+      "save"
+      "[FILENAME]"
```

Comment:
`:save` also support saving a specific expression. Would the following argument syntax work?

```suggestion
      "[FILENAME | FILENAME = EXPRESSION]"
```